holoviews/plotting/bokeh/raster.py

```diff
@@ -83,6 +84,22 @@ def get_data(self, element, ranges, style):
             key = 'image' if i == 2 else dimension_sanitizer(vdim.name)
             data[key] = [img]
 
+            # check if there's an actual value (not np.nan)
+            nanmin = np.nanmin(img)
+            nanmax = np.nanmax(img)
+            if nanmin == nanmax and clim_undefined and self.colorbar:
```

Comment:
Not sure this is the best behavior, I think it should simply add some default padding to the range.

Reply:
Also don't think this is the correct place for this logic, it should be moved to the appropriate place on ColorbarPlot.

Reply:
Right; I think it's better to have an empty plot than to change the behavior or have a warning or error, so that it will behave well in streaming or batch situations where there may legitimately be no data in that particular instance.